core/server/master/src/main/java/alluxio/master/file/RpcContext.java

```diff
@@ -126,7 +143,10 @@ public void close() throws UnavailableException {
     // JournalContext is closed before block deletion context so that file system master changes
     // are written before block master changes. If a failure occurs between deleting an inode and
     // remove its blocks, it's better to have an orphaned block than an inode with a missing block.
-    closeQuietly(mJournalContext);
+    if (mFlushJournal) {
+      // Close the journal context to reclaim resource and flush journal
+      closeQuietly(mJournalContext);
+    } // Otherwise rely on the caller to close and flush later
```

Comment:
there are cases where components will directly call rpcContext.getJournalContext().flush() ? is there anywhere in delete() or free() will render  the batch flush op moot?

Reply:
See my reply in https://github.com/Alluxio/alluxio/pull/16970#discussion_r1133190187